x-pack/plugins/maps/public/classes/layers/create_tile_map_layer_descriptor.ts

```diff
@@ -0,0 +1,149 @@
+import { VectorStyle } from '../styles/vector/vector_style';
+// @ts-ignore
+import { ESGeoGridSource } from '../sources/es_geo_grid_source';
+import { VectorLayer } from './vector_layer/vector_layer';
+// @ts-ignore
+import { HeatmapLayer } from './heatmap_layer/heatmap_layer';
+import { getDefaultDynamicProperties } from '../styles/vector/vector_style_defaults';
+import { NUMERICAL_COLOR_PALETTES } from '../styles/color_palettes';
+import { getSourceAggKey } from '../../../common/get_agg_key';
+import { isMetricCountable } from '../util/is_metric_countable';
+
+const defaultDynamicProperties = getDefaultDynamicProperties();
+
+function isHeatmap(mapType: string): boolean {
+  return mapType.toLowerCase() === 'heatmap';
+}
+
+function getGeoGridRequestType(mapType: string): RENDER_AS {
+  if (isHeatmap(mapType)) {
+    return RENDER_AS.HEATMAP;
+  }
+
+  if (mapType.toLowerCase() === 'shaded geohash grid') {
+    return RENDER_AS.GRID;
+  }
+
+  return RENDER_AS.POINT;
+}
+
+export function createAggDescriptor(
+  mapType: string,
+  metricAgg: string,
+  metricFieldName?: string
+): AggDescriptor {
+  const aggTypeKey = Object.keys(AGG_TYPE).find((key) => {
+    return AGG_TYPE[key as keyof typeof AGG_TYPE] === metricAgg;
+  });
+  const aggType = aggTypeKey ? AGG_TYPE[aggTypeKey as keyof typeof AGG_TYPE] : undefined;
+
+  return aggType && metricFieldName && (!isHeatmap(mapType) || isMetricCountable(aggType))
+    ? { type: aggType, field: metricFieldName }
+    : { type: AGG_TYPE.COUNT };
+}
+
+export function createTileMapLayerDescriptor({
+  label,
+  mapType,
+  colorSchema,
+  indexPatternId,
+  geoFieldName,
+  metricAgg,
+  metricFieldName,
+}: {
+  label: string;
+  mapType: string;
+  colorSchema: string;
+  indexPatternId?: string;
+  geoFieldName?: string;
+  metricAgg: string;
+  metricFieldName?: string;
+}): LayerDescriptor | null {
+  if (!indexPatternId || !geoFieldName) {
+    return null;
+  }
+
+  const metricsDescriptor = createAggDescriptor(mapType, metricAgg, metricFieldName);
+  const geoGridSourceDescriptor = ESGeoGridSource.createDescriptor({
+    indexPatternId,
+    geoField: geoFieldName,
+    metrics: [metricsDescriptor],
+    requestType: getGeoGridRequestType(mapType),
+    resolution: GRID_RESOLUTION.MOST_FINE,
+  });
+
+  if (isHeatmap(mapType)) {
+    return HeatmapLayer.createDescriptor({
+      label,
+      sourceDescriptor: geoGridSourceDescriptor,
+    });
+  }
+
+  const metricSourceKey = getSourceAggKey({
+    aggType: metricsDescriptor.type,
+    aggFieldName: metricsDescriptor.field,
+  });
+  const metricStyleField = {
+    name: metricSourceKey,
+    origin: FIELD_ORIGIN.SOURCE,
+  };
+
+  const colorPallette = NUMERICAL_COLOR_PALETTES.find((pallette) => {
+    return pallette.value.toLowerCase() === colorSchema.toLowerCase();
+  });
+  const styleProperties: VectorStylePropertiesDescriptor = {
+    [VECTOR_STYLES.FILL_COLOR]: {
+      type: STYLE_TYPE.DYNAMIC,
+      options: {
```

Comment:
for non-countable metrics, consider turning _off_ the "calculate color ramp range from indices"-switch.

![image](https://user-images.githubusercontent.com/1833023/94272397-01ec1f00-ff11-11ea-9723-aacb07e4173a.png)

It will expand the color-ramp, and the visual effect of the colors will be closer to the choropleth in the region-map-viz. (same for coordinate-map cluster/grids)

Reply:
I have updated vector styles to set `fieldMetaOptions.isEnabled` to false to turn off field meta